src/emitterutils.cpp

```diff
@@ -225,20 +225,34 @@ bool IsValidLiteralScalar(const std::string& str, FlowType::value flowType,
   return true;
 }
 
-void WriteDoubleQuoteEscapeSequence(ostream_wrapper& out, int codePoint) {
+std::pair<uint16_t, uint16_t> EncodeUTF16SurrogatePair(int codePoint) {
+  const uint32_t leadOffset = 0xD800 - (0x10000 >> 10);
+
+  return {
+    leadOffset | (codePoint >> 10),
+    0xDC00 | (codePoint & 0x3FF),
+  };
+}
+
+void WriteDoubleQuoteEscapeSequence(ostream_wrapper& out, int codePoint, bool escapeAsJson) {
```

Comment:
All of these helper functions would probably read better with the StringEscaping enum, rather than a boolean.

Reply:
Done.